### ISSUE TITLE
[A11y][ui5-list]: Checkbox accessible names are not unique and do no match label

### ISSUE THREAD

**Issue by caroline-ferri** [SAP/ui5-webcomponents#10666]
### Bug Description

This issue has previously been brought up in https://github.com/SAP/ui5-webcomponents/issues/6861 

For a list that has select mode set to multiple the checkboxes accessible names are not unique and do not match the physical labels on the list. This is in violation of [ACC264.3](https://wiki.one.int.sap/wiki/display/Acc/ACC-264+Details#ACC264Details-acc2643:~:text=ACC%2D264.3%3A%20Does%20the%20programmatically%20determined%20label%20name%20map%20the%20visible%20label%20text%3F)

### Affected Component

UI5-List

### Expected Behaviour

When using AT like JAWS we expect the checkboxes to be announced with unique accessible names. Additionally, these names should match the visual label used to label the checkbox.

### Isolated Example

https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/data-display-list--docs

### Steps to Reproduce

1. Using JAWS navigate to https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/data-display-list--docs
2. set selection mode to "Multiple"
3. tab to the list items
4. notice that announcement for the checkboxes lack description and clarity
...

### Priority

None

### UI5 Web Components Version

latest

### Browser

Chrome

### Operating System

Mac OS

### Additional Context

This issue has already been discussed in https://github.com/SAP/ui5-webcomponents/issues/6861 which has been closed. However, important information has been added to the issue since it has been closed. Hence, I am opening this issue. Please refer to original issue's conversation for additional context which is pertinent to the issue at hand. 

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by petyabegovska** [comment]
Hello @yanaminkova, @dobrinyonkov,

Please check the latest two comments from @judith9209 in the linked issue: https://github.com/SAP/ui5-webcomponents/issues/6861 and provide your advice.

Best regards,
Petya

**Comment by caroline-ferri** [comment]
@petyabegovska  as an addition to this issue, I want to note that this issue also persists for radio buttons in a ui5-list. When we set selection mode to "Single Start" and use the JAWS key combo "INSERT + CNTRL + A" a list on identical names are provided for both radio buttons and checkboxes. This is still occurring in latest version (2.9.0). See image below for identical radio button names. 

<img width="697" alt="Image" src="https://github.com/user-attachments/assets/34792ac9-48c0-4db1-b486-f02c836a4b32" />

**Comment by judith9209** [comment]
These are the inner source tickets,  complains from big customers are linked: 
FIORITECHE1-8933
FIORITECHP1-33005
WCAG has confirmed validity with https://github.com/w3c/wcag/discussions/4190 and even than your designers set this to obsolete. This is so frustrating and I am disappointed that such issues are still not acknowledged from UI5.
